### PR TITLE
59 multi threading

### LINKED ISSUE #59: Implement multi-threading
# Goal

Ensure different components of the system can run in a multi-threaded manner

# Additional information

- Ensure blocks are split into appropriate threads
- Ensure appropriate locks are in place

### PR DESCRIPTION
**Pull request by miklasr**
# Details

- Add multithreading
- Add demo with live camera input and live output

# Screenshots/Gif (if appropriate)

*TBC*

# Checklist

## Code
- [ ] ~I have added some tests~
- [x] I have run the tests locally to ensure they all pass by running `.scripts/build.sh -enable-testing`

## Documentation
- [x] I have updated documentation appropriately (Doxygen)
- [x] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

Closes #59 #57

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -16,110 +16,39 @@
...
-                           static_cast<int>(body_part.y * imageHeight)),
-                 circleRadius, red, -1);
-    }
-  }
+MainWindow* main_window_ptr;
 
-  // Save the image with detected points
-  cv::imwrite("./testimg.jpg", originalImage);
+void frame_callback(PostureEstimating::PoseStatus pose_status,
+                    cv::Mat input_image) {
```
Comment (Ashwin-MJ) [inline src/main.cpp:37]: Do we need the `input_image` here just now? Or was this just future-proofing for the next steps?

**Review by Ashwin-MJ — APPROVED** [review]
Really nice stuff!! Looks really neat as well and very easy to follow. Just one minor question but happy to approve

**Comment by C-Begley** [comment]
Seems to be working. However, it doesn't appear to be very accurate. It is regularly picking up body parts that aren't there

File: src/main.cpp (hunk 1 above)
Comment (miklasr) [reply src/main.cpp:37]: Just for the future

**Review by Andrew-Ritchie — APPROVED** [review]
Looks great, good work! :)

**Pull request merged**